ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/PullQueryPublisher.java

```diff
@@ -109,7 +110,11 @@ public synchronized void subscribe(final Subscriber<Collection<StreamedRow>> sub
 
     PullQueryResult result = null;
     try {
-      result = ksqlEngine.executePullQuery(
+      final ImmutableAnalysis analysis = ksqlEngine
```

Comment:
why not pass the analysis into the constructor instead of doing it again here? it looks like you are creating it before creating the publisher in `QueryEndpoint`. Does it get lost before getting here? (this might be a different code path that I'm missing)

Reply:
I don't think there's anywhere before this point that we do compute the analysis. That said, we can certainly push it up more and further consolidate calls to the analyzer. I'll give it a shot.